src/main/java/store/reduct/model/bucket/Bucket.java

```diff
@@ -283,7 +283,7 @@ public Iterator<Record> query(String entryName, Integer start, Integer stop, Int
      * @throws ReductException
      * @throws IllegalArgumentException
      */
-    Stream<Record> getRecords(@NonNull String entryName, Long queryId) throws ReductException, IllegalArgumentException {
+    Collection<Record> getRecords(@NonNull String entryName, Long queryId) throws ReductException, IllegalArgumentException {
```

Comment:
you can't return a collection.  If a user requests few gigabytes of data, you run our of memory. It must be an iterator.

Reply:
Got it